slick-testkit/src/test/scala/slick/test/jdbc/hikaricp/SlickInUseCountTest.scala

```diff
@@ -84,7 +85,11 @@ class SlickInUseCountTest extends AsyncTest[JdbcTestDB] {
     val threadPoolExecutor = threadPoolExecutorField.get(asyncExecutor)
 
     val queue = threadPoolExecutor.getClass.getMethod("getQueue").invoke(threadPoolExecutor)
-    val inUseCountField = queue.getClass.getDeclaredField("slick$util$ManagedArrayBlockingQueue$$inUseCount")
+    val inUseCountField = Seq("inUseCount", "slick$util$ManagedArrayBlockingQueue$$inUseCount").collectFirst{name =>
+	Try(queue.getClass.getDeclaredField(name)) match{
+            case Success(field) => field
```

Comment:
Yes it does ```Seq.collectFirst``` finds the first element of the traversable or iterator for which the given partial function is defined, and applies the partial function to it.

If the field can't be found then a ```java.util.NoSuchElementException``` exception will be thrown  by ```None.get```

```scala
scala> Seq.fill(10)(0).collectFirst{x => Try{10/x} match {
     | case Success(x) => x
     | }}
res9: Option[Int] = None

```

https://www.scala-lang.org/api/current/scala/collection/Seq.html#collectFirst[B](pf:PartialFunction[A,B]):Option[B]